packages/bsky/src/api/app/bsky/graph/getKnownFollowers.ts

```diff
@@ -0,0 +1,119 @@
+import { mapDefined } from '@atproto/common'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../../../lexicon'
+import { QueryParams } from '../../../../lexicon/types/app/bsky/graph/getKnownFollowers'
+import AppContext from '../../../../context'
+import {
+  HydrationFnInput,
+  PresentationFnInput,
+  RulesFnInput,
+  SkeletonFnInput,
+  createPipeline,
+} from '../../../../pipeline'
+import { HydrateCtx, Hydrator } from '../../../../hydration/hydrator'
+import { Views } from '../../../../views'
+import { clearlyBadCursor, resHeaders } from '../../../util'
+
+export default function (server: Server, ctx: AppContext) {
+  const getKnownFollowers = createPipeline(
+    skeleton,
+    hydration,
+    noBlocks,
+    presentation,
+  )
+  server.app.bsky.graph.getKnownFollowers({
+    auth: ctx.authVerifier.standard,
+    handler: async ({ params, auth, req }) => {
+      const viewer = auth.credentials.iss
+      const labelers = ctx.reqLabelers(req)
+      const hydrateCtx = await ctx.hydrator.createContext({
+        labelers,
+        viewer,
+      })
+
+      const result = await getKnownFollowers(
+        { ...params, hydrateCtx: hydrateCtx.copy({ viewer }) },
+        ctx,
+      )
+
+      return {
+        encoding: 'application/json',
+        body: result,
+        headers: resHeaders({ labelers: hydrateCtx.labelers }),
+      }
+    },
+  })
+}
+
+const skeleton = async (input: SkeletonFnInput<Context, Params>) => {
+  const { params, ctx } = input
+  const [subjectDid] = await ctx.hydrator.actor.getDidsDefined([params.actor])
+  if (!subjectDid) {
+    throw new InvalidRequestError(`Actor not found: ${params.actor}`)
+  }
+  if (clearlyBadCursor(params.cursor)) {
+    return { subjectDid, knownFollowers: [], cursor: undefined }
+  }
+
+  const res = await ctx.hydrator.dataplane.getFollowsFollowing({
+    actorDid: params.hydrateCtx.viewer,
+    targetDids: [subjectDid],
+  })
+  const result = res.results[0]
```

Comment:
This helps the type system acknowledge the case that there's no result, which I see you're handling below:
```suggestion
  const result = res.results.at(0)
```

Reply:
Oh good call, v nice